sql/cagg_utils.sql

```diff
@@ -120,3 +120,38 @@ CREATE OR REPLACE FUNCTION _timescaledb_internal.invalidation_process_cagg_log(
     OUT ret_window_start BIGINT,
     OUT ret_window_end BIGINT
 ) RETURNS RECORD AS '@MODULE_PATHNAME@', 'ts_invalidation_process_cagg_log' LANGUAGE C STRICT VOLATILE;
+
+--Get the maximum value materialized for the given hypertable
+CREATE OR REPLACE FUNCTION timescaledb_experimental.get_materialization_threshold(
+   mat_ht varchar,
+   time_col varchar
+)  RETURNS varchar
+LANGUAGE 'plpgsql'
+ AS    $func$
+ DECLARE mat_thresh TIMESTAMP;
+         query_string text;
+BEGIN
+    query_string = format('SELECT max(%s) FROM %s', time_col, mat_ht);
+    EXECUTE query_string into mat_thresh;
+    RETURN mat_thresh;
+END;
+$func$;
+
+--Function to materialize the hypertable --mat_ht using join definition in join_view
+--using the target hypertable target_ht over the provided target_thresh
+CREATE OR REPLACE FUNCTION timescaledb_experimental.execute_materialization(
```

Comment:
This isn't one of the APIs we've talked about...and I'm not sure that it's what we want. I don't think we're doing the right level of abstraction here.